docs/internals/protocol/dataformats.rst

```diff
@@ -53,6 +59,47 @@ Note: zero-length arrays (and sets) are represented as a 12-byte value where
 ``dims`` equal to zero regardless of the shape in type descriptor.
 
 
+Sets of arrays are a special case and are represented as the following
```

Comment:
I'm not sure it's easy to find out what is `Envelope` and what is `Element` and what is the difference. So some intro or comment should state something like "every element of the set of arrays is wrapped into an envelope".

It's easier to think about it like "every array within a set is wrapped into a tuple of size one", but I don't think it fits the narrative here.

Reply:
changed